pkg/controller/v1alpha1/inferencegraph/controller.go

```diff
@@ -90,7 +90,17 @@ type RouterConfig struct {
 		operations on headers. For example: Similar to "propagate" operation, one can add "transform" operation if they
 		want to transform headers keys or values before passing down to nodes.
 	*/
-	Headers map[string][]string `json:"headers"`
+	Headers          map[string][]string `json:"headers"`
+	ImagePullPolicy  string              `json:"imagePullPolicy"`
+	ImagePullSecrets []string            `json:"imagePullSecrets"`
+}
+
+func (rc *RouterConfig) GetImagePullSecrets() []v1.LocalObjectReference {
+	var imagePullSecrets []v1.LocalObjectReference
+	for _, secret := range rc.ImagePullSecrets {
+		imagePullSecrets = append(imagePullSecrets, v1.LocalObjectReference{Name: secret})
+	}
+	return imagePullSecrets
```

Comment:
shouldn't we validate it before returning it?
I mean, it just searches for a secret, if the user accidentally refers other a secret other than one that contains the registry credentials it would fail to pull the images.